### ISSUE TITLE
Wrong val loss term in sentiment classification

### ISSUE THREAD

**Issue by HowsenH** [pytorch/tutorials#913]
In https://github.com/pytorch/tutorials/blob/master/beginner_source/text_sentiment_ngrams_tutorial.py the evaluation code piece overwrites loss everytime to be the loss of the last batch, and by averaging them in the end we have an unreasonably small loss.

```
def test(data_):
    loss = 0
    acc = 0
    data = DataLoader(data_, batch_size=BATCH_SIZE, collate_fn=generate_batch)
    for text, offsets, cls in data:
        text, offsets, cls = text.to(device), offsets.to(device), cls.to(device)
        with torch.no_grad():
            output = model(text, offsets)
            loss = criterion(output, cls)             <- should use a different name
            loss += loss.item()
            acc += (output.argmax(1) == cls).sum().item()

    return loss / len(data_), acc / len(data_)
```

**Comment by holly1238** [comment]
This is no longer an issue in the current version of the code.
